instrumentation/trilogy/lib/opentelemetry/instrumentation/trilogy/patches/client.rb

```diff
@@ -87,10 +87,14 @@ def obfuscate_sql(sql)
             if sql.size > 2000
               'SQL query too large to remove sensitive data ...'
             else
-              obfuscated = sql.gsub(FULL_SQL_REGEXP, '?')
+              obfuscated = OpenTelemetry::Common::Utilities.utf8_encode(sql, binary: true)
+              obfuscated = obfuscated.gsub(FULL_SQL_REGEXP, '?')
               obfuscated = 'Failed to obfuscate SQL query - quote characters remained after obfuscation' if detect_unmatched_pairs(obfuscated)
               obfuscated
             end
+          rescue StandardError => e
+            OpenTelemetry.logger.error("Failed to obfuscate sql: #{e.message}")
+            nil
```

Comment:
Lets keep the default message in addition to the error handling Ariel suggested above. The reasoning here is that if we default to nil, that attribute will get dropped when the span goes through its finalization steps. I think dropping the opentelemetry error as an attribute value on the span is a little bit on the crude side, but if a user is looking at a trace and sees a DB span with this message on the `db.statement` at least they have some sense that something went wrong, with nil it will just be absent which is a bit more confusing IMO.
```suggestion
            'OpenTelemetry error: failed to obfuscate sql'
```

Reply:
Got it! Thanks for clarifying